microsoft-edge/webview2/concepts/frames.md

```diff
@@ -0,0 +1,496 @@
+
+URLs that aren't in the `origins` list will need to be implemented on the native side of the host app.
+
+##### [.NET/C#](#tab/dotnetcsharp)
+
+* [CoreWebView2Frame.AddHostObjectToScript Method](/dotnet/api/microsoft.web.webview2.core.corewebview2frame.addhostobjecttoscript) - has an `origins` parameter.  The documentation doesn't have a **Method name** table.
+
+The above method works like the following method:
+* [CoreWebView2.AddHostObjectToScript Method](/dotnet/api/microsoft.web.webview2.core.corewebview2.addhostobjecttoscript).  See the **Method name** table.  Read both of these API Reference topics, although for frames, you would use the method which supports an `origins` parameter instead.
+
+##### [WinRT/C#](#tab/winrtcsharp)
+
+<!-- TODO -->
+Not implemented yet.
+
+<!-- * [CoreWebView2Frame.AddHostObjectToScript Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2frame#addhostobjecttoscript) - has an `origins` parameter.
+* [CoreWebView2Frame.RemoveHostObjectFromScript Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2frame#removehostobjectfromscript) -->
+
+<!-- The above method works like the following method: -->
+<!-- * [CoreWebView2.AddHostObjectToScript Method](/microsoft-edge/webview2/reference/winrt/microsoft_web_webview2_core/corewebview2#addhostobjecttoscript).  See the **Method name** table.  Read both of these API Reference topics, although for frames, you would use the method which supports an `origins` parameter instead. -->
+
+##### [Win32/C++](#tab/win32cpp)
+
+* [ICoreWebView2Frame::AddHostObjectToScriptWithOrigins method](/microsoft-edge/webview2/reference/win32/icorewebview2frame#addhostobjecttoscriptwithorigins) - has an `origins` parameter.  The documentation doesn't have a **Method name** table.
+
+The above method works like the following method:
+* [ICoreWebView2::AddHostObjectToScript method](/microsoft-edge/webview2/reference/win32/icorewebview2#addhostobjecttoscript).  See the **Method name** table.  Read both of these API Reference topics, although for frames, you would use the method which supports an `origins` parameter instead.
+
+---
+
+<!-- ------------------------------ -->
+#### Example code
+
+See [Step 6: Call AddHostObjectToScript to pass the host object to web-side code](../how-to/hostobject.md#step-6-call-addhostobjecttoscript-to-pass-the-host-object-to-web-side-code) in _Call native-side code from web-side code_.
+
+
+See also:
+* [Host/web object sharing](./overview-features-apis.md#hostweb-object-sharing) in _Overview of WebView2 features and APIs_.
+
+
+<!-- ====================================================================== -->
+## Sending and receiving messages
+
+You can send and receive messages between native app and JavaScript code in iframe.  Messages are sent from iframe in an HTML page (via JavaScript) and are received by the host app.
```

Comment:
_Messages are sent from iframe in an HTML page (via JavaScript) and are received by the host app._ - they are two dirrectional, may add `or an opposite direction` or `or vise versa`

Reply:
formed 2 parallel statements, opposite direction